### ISSUE TITLE
Multiple Check-Ins for the same Event (QR-Code) are possible

### ISSUE THREAD

**Issue by jkrwdf** [corona-warn-app/cwa-app-android#2932]
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [ ] Bug is specific for Android only (cannot identify, have only Android)
* [X] Bug is not already reported in another issue

## Describe the bug
I can check-in multiple times to the same event, represented by its QR code. 
This is obviously not sensible and could be avoided.
See attached screenshot. 

![image](https://user-images.githubusercontent.com/5329491/115878965-70fb0900-a449-11eb-98c2-3f8a8cec756b.png)

## Steps to reproduce the issue
Create QR code on one device. 
Check in multiple times using another device.

## Expected behaviour
I think, re-scanning the same QR code while a check-in for the same QR code is still active should update the existing check-in and prolong its check-in time by the "default check-in time" of the event. 

Example: Default check-in time is 2 hours. 
I checked in 1 hour ago. I now check in again. 
Result: I still have only one active checkin, and it will automatically check out two hours from now.

Less good would be an error message "You are already checked in here".

**Comment by Ein-Tim** [comment]
Confirm same bug under iOS.

**Comment by Ein-Tim** [comment]
@jkrwdf I suggest that you open a separate feature request for this in the https://github.com/corona-warn-app/cwa-wishlist repository.

**Comment by thomasaugsten** [comment]
There also some person which visit multiple rooms on an event at the same time. For example waiters or visitor on conferences. We don't want to restrict here to much.

**Comment by vaubaehn** [comment]
How about if scanning same QR code again, the automatic check out time is added to the remaining time of that event/location?

**Comment by jkrwdf** [comment]
@vaubaehn I think this suggestion is not as realistic as the approach to set the checkout time of the event in that case to the current time plus 1 times the automatic checkout time. In my initial example, I would have 6 hours otherwise.

**Comment by vaubaehn** [comment]
@jkrwdf 
> @vaubaehn I think this suggestion is not as realistic as the approach to set the checkout time of the event in that case to the current time plus 1 times the automatic checkout time. In my initial example, I would have 6 hours otherwise.

True, but I can imagine different use cases, where I would find it comfortable for myself.
What about, if CWA asked you what to do, if you scanned the same QR code again while still being checked in?
Possible options:
-> cancel (dismiss scan/don't check in again)
-> add (automatic checkout) time to remainng time
-> modify check in time/remaining time manually (while still being checked in)

If someone found this useful, I'm not sure if this is a bug fix or an enhancement request...
